### ISSUE TITLE
Grafana - switch to file-based dashboard provisioning

### ISSUE THREAD

**Issue by kmoppel** [cybertec-postgresql/pgwatch2#61]
From Grafana 5.0 it's possible to declare files that should be loaded. Currently we're inserting them directly into DB.

http://docs.grafana.org/administration/provisioning/#dashboards

**Comment by marmelin** [comment]
i would appreciate that, because we have problems with the import into database.

**Comment by kmoppel** [comment]
@marmelin ok, can you maybe specify a little - what kind of issue? maybe there's some intermediate fix....

**Comment by marmelin** [comment]
thx @kmoppel ... I am running pgwatch2 on an openshift environment. After startup the system is running, but only shows overview values - as soon as I click on a button in order to get detailed view, I am getting an error which says "Dashboard not found".

So I looked up "dashboard" table in database and it shows me only 18 entries:
```
pgwatch2_grafana=> select id,version,slug,title from dashboard;
 id | version |            slug            |           title
----+---------+----------------------------+----------------------------
  1 |       1 | documentation              | Documentation
  2 |       1 | global-db-overview         | Global DB Overview
  3 |       1 | health-check               | Health-check
  4 |       1 | index-overview             | Index overview
  6 |       1 | pgbouncer-stats            | PgBouncer stats
  7 |       1 | postgres-version-overview  | Postgres Version Overview
  8 |       1 | replication                | Replication lag
  9 |       1 | single-query-details       | Single query details
 10 |       1 | sproc-details              | Sproc details
 11 |       1 | sprocs-top                 | Sprocs Top
 12 |       1 | stat-statements-overview   | Stat statements overview
  5 |       1 | lock-details               | Lock details
 13 |       1 | stat-statements-sql-search | Stat Statements SQL Search
 14 |       1 | stat-statements-top        | Stat statements Top
 15 |       1 | stat_activity_realtime     | Stat Activity Realtime
 16 |       1 | system-stats               | System Stats
 17 |       1 | table-details              | Table details
 18 |       1 | tables-top                 | Tables top
(18 rows)
```
However, I edited the "import_dashboard.sh" script and I got 24 dashboard files. So I ran the script by myself and ended up in having double entries for the dashboards ;) ... so nothing worked then. In order to resolv I deleted all existing entries and imported again. This time only 24 entries are there, but I does not show any stats, because I am having something like "influxdb entry" error.: 
```
pgwatch2_grafana=> select id,title,slug from dashboard;
 id |                   title                   |            slug
----+-------------------------------------------+-----------------------------
  1 | AWS CloudWatch overview                   | aws-cloudwatch
  2 | Biggest relations treemap                 | biggest-relations
  3 | DDL / Config change events                | change-events
  4 | Checkpointer / Bgwriter / Block IO Stats  | checkpointer-bgwriter-stats
  5 | DB overview                               | db-overview
  6 | DB overview Unprivileged / Developer mode | db-overview-developer
  7 | Documentation                             | documentation
  8 | Global DB Overview                        | global-db-overview
  9 | Health-check                              | health-check
 10 | Index overview                            | index-overview
 11 | Lock details                              | lock-details
 12 | PgBouncer stats                           | pgbouncer-stats
 13 | Postgres Version Overview                 | postgres-version-overview
 14 | Replication lag                           | replication
 15 | Single query details                      | single-query-details
 16 | Sproc details                             | sproc-details
 17 | Sprocs Top                                | sprocs-top
 18 | Stat statements overview                  | stat-statements-overview
 19 | Stat Statements SQL Search                | stat-statements-sql-search
 20 | Stat statements Top                       | stat-statements-top
 21 | System Stats                              | system-stats
 22 | Table details                             | table-details
 23 | Tables top                                | tables-top
(23 rows)
```

Long story short - I think if grafana simply has to read files from a folder only, such problems would have been gone.

**Comment by marmelin** [comment]
Can you tell, when pgwatch2 will support grafana dashboards with files ?

**Comment by kmoppel** [comment]
@marmelin hugh, a tough question. no real timeline for that as currently it also "works" :)
